### ISSUE TITLE
DHCP client in bridged mode

### ISSUE THREAD

**Issue by willat8** [seud0nym/tch-gui-unhide#113]
Hi @seud0nym, would it be possible to configure proto dhcp for the lan interface when in bridge mode, via the UI?
It works well if I do `uci set network.lan.proto='dhcp'`, `uci delete network.lan.ipaddr` and `uci delete network.lan.netmask`, but this causes the Local Network card to not work anymore.

**Comment by seud0nym** [comment]
So you want to assign the LAN IP address of the device via DHCP? Sounds very problematic. The bridged device will end up with a WAN IP address provided by your RSP, in addition to the one that your downstream router gets from the RSP. Would that even work? And your bridge then has a public IP with no firewall to protect it.

I don't think this is a good idea...

**Comment by willat8** [comment]
In bridged mode the WAN interface is part of the bridge too, so wouldn't the only difference be how the bridge gets its IP address assigned? It sounds like a bad idea to bridge the NTD anyway, assuming you could even get multiple IPs from the RSP.

My use-case is bridge client mode, where I can give my second DJA0230 a static lease on from primary, and use it as either a WLAN bridge or an ethernet bridge (or in my case just an SMB share).

**Comment by seud0nym** [comment]
Okay, sorry, I misinterpreted the meaning of "DHCP client in bridged mode".

I set up a static lease in my primary and assign the same static IP on the device, but I guess you are thinking about not having to update the device if you ever change its IP?

I will have to think about it. The Local Network screen is very complicated and large and slow already.

**Comment by willat8** [comment]
Yep :) Thanks!

**Comment by seud0nym** [comment]
Well, it was a little bit painful, but it is done. You can download and install the pre-release test version with:
```
./tch-gui-unhide -u
./tch-gui-unhide -y
```

You only get the option of Static or DHCP in bridged mode.

**Comment by willat8** [comment]
Thank you for this!

Works beautifully, but seems like the IPv6 State toggle is not having an effect, it remains on.

**Comment by seud0nym** [comment]
So you have a static IP and IPv6 on, set it to DHCP and IPv6 off, and when you go to the new IP address (or refresh if it is the same IP), IPv6 is still on? Very strange. I cannot reproduce that.

**Comment by willat8** [comment]
Currently it shows
![Screenshot_20221013-174748~2](https://user-images.githubusercontent.com/2316013/195522635-56818b33-27f0-49b5-a4a8-30c29fcb541e.png)

Then when I toggle it off I get
![Screenshot_20221013-174905](https://user-images.githubusercontent.com/2316013/195522716-b6af768c-b88c-4e6d-8bc1-983cbc1653f8.png)

Then when I refresh it's back to On.

**Comment by willat8** [comment]
Looks like was caused by those original `uci` commands I posted. Swapping to static and then back to dhcp in the UI causes everything to work properly.

Very happy customer :)

**Comment by seud0nym** [comment]
Good to hear. I couldn't work out which parameter was failing validation...

**Comment by willat8** [comment]
I wonder if this could also be added as a switch in the wifi booster script (to obtain an address from the primary device via dhcp instead of specifying manually/defaulting to the current address or 192.168.0.2)?

**Comment by seud0nym** [comment]
Fixed in Release 2022.11.01

**Comment by willat8** [comment]
I've just tried this on 18.1.c. After enabling DHCP in bridged mode, I don't get a default route anymore. After manually adding it with `ip route add default via 192.168.0.1` everything is fine, until the next reboot.

Any ideas? Everything was fine on 20.3.c.

**Comment by seud0nym** [comment]
It looks like the default for the DHCP options that are requested from the DHCP server changed between 18.1.c and 20.3.c. I have now changed it to explicitly request the things that are needed, including the default gateway which in turn will generate the default route.

I have also fixed the IP address in content and printer sharing. It was using the configured static IP rather than the actual DHCP acquired IP.

These changes have been applied in tch-gui-unhide and bridged-booster. There is a new pre-release available for tch-gui-unhide. You can update with this command:
```
./tch-gui-unhide -u && ./tch-gui-unhide -y
```

The bridged-booster script doesn't have auto-update capability, so you will need to download it manually:
```
curl -skLo bridged-booster https://raw.githubusercontent.com/seud0nym/tch-gui-unhide/master/wifi-booster/bridged-booster
chmod +x bridged-booster
```

**Comment by willat8** [comment]
As always - thank you.

I've tried applying the 14:32 pre-release, but it's not setting the `reqopts` when running tch-gui-unhide, do I need to toggle DHCP mode off and on in the GUI as well?

**Comment by seud0nym** [comment]
Sorry, I should have mentioned that. Yes, you will need to toggle DHCP. It only applies it on change to DHCP.

Also, thank you for the donation. Much appreciated.

**Comment by willat8** [comment]
Both fixes work wonderfully. And welcome.

**Comment by willat8** [comment]
I've tried this from a fresh install on today's release (on 18.1.c), and the first time I swapped from static to dhcp in bridged mode, the reqopts didn't get added. I had to swap back to static and then back to dhcp for it to get set correctly.

**Comment by seud0nym** [comment]
Hmmm. I can't reproduce this, and I can't see how it would happen. Every time the Save button is pressed in Bridged mode, it checks to see if the proto option is dhcp, and if so, sets the reqopts. 

However, if an error does occur when saving, it is silently ignored at the moment. So, I have added a check so that any errors that occur in the save process will be displayed.

**Comment by willat8** [comment]
Sounds good.

When this happened to me, I was configuring via a direct connection to a computer with static IPs (to avoid any internet access before de-telstra) so when the original swap to DHCP happened, the DJA0230 wouldn't have been able to get an address until I powered it down and plugged it in to the rest of the network. Not sure if that contributed.

**Comment by seud0nym** [comment]
Fixed in release 2022.12.21
